src/main/java/org/automation/utilities/WebdriverWaits.java

```diff
@@ -140,4 +143,45 @@ public static void SwitchToNewTab() throws InterruptedException {
 
 		Thread.sleep(3000);
 	}
+
+    public static void retryClick(By locator, int retryCount) {
+        int waitTime = 5;
+        if (retryCount <= 0) {
+            throw new IllegalArgumentException("Retry count must be greater than 0");
+        }
+
+        int attempts = 0;
+        boolean isClicked = false;
+
+        while (attempts < retryCount && !isClicked) {
+            try {
+                /* Wait until the element is clickable */
+
+                WebDriverWait wait = new WebDriverWait(getDriver(), Duration.ofSeconds(waitTime));
+                WebElement element = wait.until(ExpectedConditions.elementToBeClickable(locator));
+
+                // Attempt to click the element
+                element.click();
+                isClicked = true;
+                System.out.println("Click successful on attempt " + (attempts + 1));
+            } catch (Exception e) {
+                attempts++;
+                if (attempts < retryCount) {
+                    System.out.println("Click attempt " + attempts + " failed. Retrying...");
+                } else {
+                    System.out.println("Click failed after " + attempts + " attempts. Locator: " + locator.toString());
+                    throw e; // Rethrow exception after all retries fail
+                }
+
+                // Add a delay before retrying
+                try {
+                    Thread.sleep(1000 * attempts); // Incremental delay (e.g., 1s, 2s, 3s...)
+                } catch (InterruptedException ie) {
```

Comment:
Instead of swallowing exceptions, Can log the exception with proper messages. This helps in debugging when issues arise otherwise we'll be stuck in debugging.